### DISCUSSION TITLE
Two seperate menus

### DISCUSSION THREAD

**Discussion by mxmtsk** [spatie/laravel-navigation#16] [Q&A]
I wanted to create two seperate menus in my application like this basic example:

```php
$navigation = Navigation::make()
    ->add(
        'Data',
        route('profile.show', ['user' => $user->id]),
        null,
    );

$navigation2 = Navigation::make()
    ->add(
        'Data2',
        route('profile.show', ['user' => $user->id]),
        null,
    );

var_dump($navigation2->tree());die;
```

The issue I face is, that the second menu will have the item of the first menu. Is that an issue or by design or am I simply missing something here.

**Comment by jappi00** [comment]
Hello @mxmtsk,

Did you find an answer for this? I think it is actually intentional, but a nice way to handle multiple, different menus would be cool.

Currently, I would suggest creating one section for each menu and use them after the tree is generated.
But I think we have one problem with that: We cannot mark two items as active at the same time.

```php
$navigation = Navigation::make()
    ->add('menu1', False, function (Section $section) {
        $section
            ->add('Overview', route('account.view'))
            ->add('Settings', route('account.view'))
            ->add('Secruity', route('account.view'))
            ->add('Logs', route('account.view'));
    })
    ->add('menu2', False, function (Section $section) {
        $section
            ->add('Overview', route('account.view'))
            ->add('Settings', route('account.view'))
            ->add('Secruity', route('account.view'))
            ->add('Logs', route('account.view'));
    });

var_dump($navigation->tree()['menu1']); # Menu 1

var_dump($navigation->tree()['menu2']); # Menu 2

```

 I would love to add something like this but I am not sure which way is the best.

**Comment by jappi00** [comment]
So, it seems that the navigation is stored via the make() calls, which creates a service container. 

I simply tried to remove the `make()` calls, but it takes no effect since it do the same thing here: https://github.com/spatie/laravel-navigation/blob/2082fe1b62092555f31c2508f848cf057404eede/src/NavigationServiceProvider.php#L16

I'm not certain why this make() was built in, since it get set in the service provider anyway.

So if you remove the make() calls and override the ServiceProvider to not scope the class, it should work.

Maybe we should mention it in the docs or remove it from the ServiceProvider entirely, or find a way to define multiple menus in the container.

What do you think @freekmurze? I hope I don't interrupt you with the mention. I don't know if it is the right way.

**Reply by sebastiandedeyne** [reply]
`Navigation` is registered as a singleton because most people use it for a single menu.

If you want to create multiple instances, you can create your own with `new Navigation()` instead of `Navigation::make()`. Then you can pass them along or store them however you want.

**Reply by SDIjeremy** [reply]
do you have an example of this? When I do `new Navigation()` then it asks for ActiveURLChecker

**Reply by sebastiandedeyne** [reply]
You can instantiate an object with the `build` method on Laravel's container.

```php
$navigation = app()->build(Navigation::class);
```

**Reply by ithuis** [reply]
i have multiple menu's and i extend the navigation

```
<?php

namespace App\Helpers\Menu;

use Spatie\Navigation\Navigation;

class MainMenu Extends Navigation
{
    
}

```
then add this to serviceProvider 

```
 $this->app->scoped(MainMenu::class);
```
seems to work fine
